site/publish.sh

```diff
@@ -1,26 +1,132 @@
-rsync -acvz www2/ --exclude=/updates --delete ${RSYNC_USER}@${UPDATES_SITE}:/var/www/${UPDATES_SITE}
+## 'www2' folder processing
+chmod -R a+r ./www2
+
+function parallelfunction() {
+    echo "=== parallelfunction: $1"
+
+    case $1 in
+    rsync*)
+        # Push generated index to the production server
+        time rsync --archive --checksum --verbose --compress \
+            --exclude=/updates `# populated by https://github.com/jenkins-infra/crawler` \
+            --delete `# delete old sites` \
+            --stats `# add verbose statistics` \
+            www2/ ${RSYNC_USER}@${UPDATES_SITE}:/var/www/${UPDATES_SITE}
+        ;;
+
+    azsync*)
+        # Sync Azure File Share content using www3 to avoid symlinks
+        time azcopy sync ./www3/ "https://updatesjenkinsio.file.core.windows.net/updates-jenkins-io/?${UPDATES_FILE_SHARE_QUERY_STRING}" \
+            --recursive=true \
+            --delete-destination=true
+        ;;
+
+    s3sync*)
+        # Retrieve the R2 bucket and the R2 endpoint from the task name passed as argument, minus "s3sync" prefix
+        updates_r2_bucket_and_endpoint="${1#s3sync}"
+        r2_bucket=${updates_r2_bucket_and_endpoint%|*}
+        r2_endpoint=${updates_r2_bucket_and_endpoint#*|}
+
+        # Sync CloudFlare R2 buckets content excluding 'updates' folder from www3 sync (without symlinks)
+        # as this folder is populated by https://github.com/jenkins-infra/crawler/blob/master/Jenkinsfile
+        time aws s3 sync ./www3/ s3://"${r2_bucket}"/ \
+            --no-progress \
+            --no-follow-symlinks \
+            --size-only \
+            --exclude '.htaccess' \
+            --endpoint-url "${r2_endpoint}"
+        ;;
+
+    *)
+        echo -n "Warning: unknown parameter"
+        ;;
+
+    esac
+}
+
+# Export local variables used in parallelfunction
+export UPDATES_SITE
+export RSYNC_USER
+
+# Export function to use it with parallel
+export -f parallelfunction
+
+# Sync only updates.jenkins.io by default
+tasks=("rsync")
+
+# Sync updates.jenkins.io and azure.updates.jenkins.io File Share and R2 bucket(s) if the flag is set
+if [[ $OPT_IN_SYNC_FS_R2 == "optin" ]]
+then
+    # TIME sync, used by mirrorbits to know the last update date to take in account
+    date +%s > ./www2/TIME
+
+    ## No need to remove the symlinks as the `azcopy sync` for symlinks is not yet supported and we use `--no-follow-symlinks` for `aws s3 sync`
+    # Perform a copy with dereference symlink (object storage do not support symlinks)
+    # copy & transform simlinks into referent file/dir
+    rsync --archive --checksum --verbose --compress \
+                --copy-links `# derefence symlinks` \
+                --safe-links `# ignore symlinks outside of copied tree` \
+                --stats `# add verbose statistics` \
+                --exclude='updates' `# populated by https://github.com/jenkins-infra/crawler` \
+                --delete `# delete old sites` \
+                www2/ www3/
```

Comment:
```suggestion
    rm -rf ./www3/ # Cleanup
    
    rsync --archive --verbose \
                --copy-links `# derefence symlinks` \
                --safe-links `# ignore symlinks outside of copied tree` \
                --exclude='updates' `# Exclude ALL 'updates' directories, not only the root /updates (because symlink dereferencing create additional directories` \
                ./www2/ ./www3/
```

- Cleanup `./www3` any leftover from previous execution
- Remove unneeded rsync options for this particular operation
- Update  comment for the exclude to explain "why" do we need to remove the `/` prefix (compared to the other rsync)